### PR TITLE
[QACI-231] - Created Enterprise And Community Profiles

### PR DESCRIPTION
**Pull request by cubastanley**
Update for new versioning system within the test suites - Legacy versions should be tested against the "old-payara-versioning" branch

### CODE REVIEW

File: MicroProfile-Config/tck-runner/pom.xml [hunk 1]
```diff
@@ -246,32 +246,34 @@
...
-        
+
         <profile>
-            <id>mp-config-1.3</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/4\..+|5\.19[1-4].*|5\.201.*/</value>
+                    <value>/^[5-9]\.[0-9]{4}(-SNAPSHOT|\..*)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Config/tck-runner/pom.xml:255]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-Config/tck-runner/pom.xml [hunk 2]
```diff
@@ -246,32 +246,34 @@
...
         </profile>
+
         <profile>
-            <id>mp-config-1.4</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.202.*/</value>
+                    <value>/^[5-9]\.[0-9]{2}\.[0-9]{1,2}(-SNAPSHOT)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Config/tck-runner/pom.xml:269]:
```suggestion
                    <value>/^[5-9]\.[0-9]{2}\.[0-9].*/</value>
```

File: MicroProfile-Config/tck-runner/pom.xml [hunk 3]
```diff
@@ -246,32 +246,34 @@
...
-                <microprofile.config.version>1.3</microprofile.config.version>
-                <microprofile.config.tck.version>1.3</microprofile.config.tck.version>
+                <microprofile.config.version>1.4</microprofile.config.version>
+                <microprofile.config.tck.version>1.4</microprofile.config.tck.version>
             </properties>
         </profile>
+
         <profile>
-            <id>mp-config-1.4</id>
+            <id>Payara Enterprise</id>
```
Comment (Pandrex247) [inline MicroProfile-Config/tck-runner/pom.xml:265]: While fine for Community, this probably needs more thought for Enterprise unless you want to take a "mañana" approach and cross that bridge when we come to it - It's entirely possibly that a single release of Enterprise will support two different versions of an API.

File: MicroProfile-Fault-Tolerance/tck-runner/pom.xml [hunk 4]
```diff
@@ -126,39 +126,26 @@
 
     <profiles>
         <profile>
-            <id>mp-ft-1.1</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/4\..+|5\.18.+|5\.191.*/</value>
+                    <value>/^[5-9]\.[0-9]{4}(-SNAPSHOT|\..*)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Fault-Tolerance/tck-runner/pom.xml:133]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-Fault-Tolerance/tck-runner/pom.xml [hunk 5]
```diff
@@ -126,39 +126,26 @@
...
         </profile>
+
         <profile>
-            <id>mp-ft-2.1</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>5\.20[^1].*/</value>
+                    <value>/^[5-9]\.[0-9]{2}\.[0-9]{1,2}(-SNAPSHOT)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Fault-Tolerance/tck-runner/pom.xml:148]:
```suggestion
                    <value>/^[5-9]\.[0-9]{2}\.[0-9].*/</value>
```

File: MicroProfile-Health/tck-runner/pom.xml [hunk 6]
```diff
@@ -204,28 +204,30 @@
         </profile>
 
         <profile>
-            <id>mp-hc-1.0</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/4\..+|5\.18.+|5\.19[^4].*/</value>
+                    <value>/^[5-9]\.[0-9]{4}(-SNAPSHOT|\..*)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Health/tck-runner/pom.xml:211]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-Health/tck-runner/pom.xml [hunk 7]
```diff
@@ -204,28 +204,30 @@
...
         </profile>
+
         <profile>
-            <id>mp-hc-2.1</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.194.*|5\.2.*/</value>
+                    <value>/^[5-9]\.[0-9]{2}\.[0-9]{1,2}(-SNAPSHOT)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Health/tck-runner/pom.xml:224]:
```suggestion
                    <value>/^[5-9]\.[0-9]{2}\.[0-9].*/</value>
```

File: MicroProfile-Metrics/tck-runner/pom.xml [hunk 8]
```diff
@@ -143,62 +143,29 @@
 
     <profiles>
         <profile>
-            <id>mp-metrics-1.2</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/^(4\.1\.2\.[1-9]\d{2}|5\.19[1-2])(\.\d*)?(-SNAPSHOT)?/</value>
+                    <value>/^[5-9]\.[0-9]{4}(-SNAPSHOT|\..*)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Metrics/tck-runner/pom.xml:150]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-Metrics/tck-runner/pom.xml [hunk 9]
```diff
@@ -143,62 +143,29 @@
...
-            </properties>
-        </profile>
         <profile>
-            <id>mp-metrics-2.3</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.20[^1].*/</value>
+                    <value>/^[5-9]\.[0-9]{2}\.[0-9]{1,2}(-SNAPSHOT)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Metrics/tck-runner/pom.xml:168]:
```suggestion
                    <value>/^[5-9]\.[0-9]{2}\.[0-9].*/</value>
```

File: MicroProfile-OpenAPI/tck-runner/pom.xml [hunk 10]
```diff
@@ -109,32 +109,35 @@
...
     <profiles>
+
         <profile>
-            <id>mp-openapi-1.0</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/4\..+|5\.18.+|5\.191.*/</value>
+                    <value>/^[5-9]\.[0-9]{4}(-SNAPSHOT|\..*)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-OpenAPI/tck-runner/pom.xml:118]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-OpenAPI/tck-runner/pom.xml [hunk 11]
```diff
@@ -109,32 +109,35 @@
...
         </profile>
+
         <profile>
-            <id>mp-openapi-1.1</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.19[^1].*|5\.2.+/</value>
+                    <value>/^[5-9]\.[0-9]{2}\.[0-9]{1,2}(-SNAPSHOT)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-OpenAPI/tck-runner/pom.xml:132]:
```suggestion
                    <value>/^[5-9]\.[0-9]{2}\.[0-9].*/</value>
```

File: MicroProfile-Rest-Client/tck-runner/pom.xml [hunk 12]
```diff
@@ -113,119 +113,18 @@
...
     <profiles>
-        <!-- Payara 5 versions up to 5.191 - Rest Client 1.1 -->
         <profile>
-            <id>payara5.181-191</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.1(8\d*|91)(?:-SNAPSHOT|\.\d*)?/</value>
+                    <value>/^[5-9]\.[0-9]{4}(-SNAPSHOT|\..*)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Rest-Client/tck-runner/pom.xml:121]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-Rest-Client/tck-runner/pom.xml [hunk 13]
```diff
@@ -281,12 +180,11 @@
...
         </profile>
         <profile>
-            <!-- Payara 5.202+ - Rest Client 1.4 -->
-            <id>payara5.202+</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.20[2-4].*/</value>
+                    <value>/^[5-9]\.[0-9]{2}\.[0-9]{1,2}(-SNAPSHOT)*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Rest-Client/tck-runner/pom.xml:187]:
```suggestion
                    <value>/^[5-9]\.[0-9]{2}\.[0-9].*/</value>
```

File: MicroProfile-Config/tck-runner/pom.xml (hunk 3 above)
Comment (cubastanley) [reply MicroProfile-Config/tck-runner/pom.xml:265]: For the time being I vote mañana to keep it clean while there's no difference between the two - plus, in my head if there were support for multiple versions of and API in one version of Enterprise, that activation probably wouldn't be based on Payara version anymore

File: MicroProfile-Fault-Tolerance/tck-runner/pom.xml [hunk 14]
```diff
@@ -126,39 +126,26 @@
 
     <profiles>
         <profile>
-            <id>mp-ft-1.1</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/4\..+|5\.18.+|5\.191.*/</value>
+                    <value>/^[5-9]\.[0-9]{4}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Fault-Tolerance/tck-runner/pom.xml:133]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```
Bit better

File: MicroProfile-Fault-Tolerance/tck-runner/pom.xml [hunk 15]
```diff
@@ -126,39 +126,26 @@
...
         </profile>
+
         <profile>
-            <id>mp-ft-2.1</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>5\.20[^1].*/</value>
+                    <value>/^[5-9]\.[0-9]{1,2}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Fault-Tolerance/tck-runner/pom.xml:148]:
```suggestion
                    <value>/^[5-9]\.[0-9]{1,2}\.[0-9].*/</value>
```

File: MicroProfile-Health/tck-runner/pom.xml [hunk 16]
```diff
@@ -204,28 +204,30 @@
         </profile>
 
         <profile>
-            <id>mp-hc-1.0</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/4\..+|5\.18.+|5\.19[^4].*/</value>
+                    <value>/^[5-9]\.[0-9]{4}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Health/tck-runner/pom.xml:211]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-Health/tck-runner/pom.xml [hunk 17]
```diff
@@ -204,28 +204,30 @@
...
         </profile>
+
         <profile>
-            <id>mp-hc-2.1</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.194.*|5\.2.*/</value>
+                    <value>/^[5-9]\.[0-9]{1,2}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Health/tck-runner/pom.xml:224]:
```suggestion
                    <value>/^[5-9]\.[0-9]{1,2}\.[0-9].*/</value>
```

File: MicroProfile-Metrics/tck-runner/pom.xml [hunk 18]
```diff
@@ -143,62 +143,29 @@
 
     <profiles>
         <profile>
-            <id>mp-metrics-1.2</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/^(4\.1\.2\.[1-9]\d{2}|5\.19[1-2])(\.\d*)?(-SNAPSHOT)?/</value>
+                    <value>/^[5-9]\.[0-9]{4}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Metrics/tck-runner/pom.xml:150]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-Metrics/tck-runner/pom.xml [hunk 19]
```diff
@@ -143,62 +143,29 @@
...
-            </properties>
-        </profile>
         <profile>
-            <id>mp-metrics-2.3</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.20[^1].*/</value>
+                    <value>/^[5-9]\.[0-9]{1,2}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Metrics/tck-runner/pom.xml:168]:
```suggestion
                    <value>/^[5-9]\.[0-9]{1,2}\.[0-9].*/</value>
```

File: MicroProfile-OpenAPI/tck-runner/pom.xml [hunk 20]
```diff
@@ -109,32 +109,35 @@
...
     <profiles>
+
         <profile>
-            <id>mp-openapi-1.0</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/4\..+|5\.18.+|5\.191.*/</value>
+                    <value>/^[5-9]\.[0-9]{4}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-OpenAPI/tck-runner/pom.xml:118]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-OpenAPI/tck-runner/pom.xml [hunk 21]
```diff
@@ -109,32 +109,35 @@
...
         </profile>
+
         <profile>
-            <id>mp-openapi-1.1</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.19[^1].*|5\.2.+/</value>
+                    <value>/^[5-9]\.[0-9]{1,2}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-OpenAPI/tck-runner/pom.xml:132]:
```suggestion
                    <value>/^[5-9]\.[0-9]{1,2}\.[0-9].*/</value>
```

File: MicroProfile-Rest-Client/tck-runner/pom.xml [hunk 22]
```diff
@@ -113,119 +113,18 @@
...
     <profiles>
-        <!-- Payara 5 versions up to 5.191 - Rest Client 1.1 -->
         <profile>
-            <id>payara5.181-191</id>
+            <id>Payara Community</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.1(8\d*|91)(?:-SNAPSHOT|\.\d*)?/</value>
+                    <value>/^[5-9]\.[0-9]{4}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Rest-Client/tck-runner/pom.xml:121]:
```suggestion
                    <value>/^[5-9]\.[0-9]{4}\.[0-9].*/</value>
```

File: MicroProfile-Rest-Client/tck-runner/pom.xml [hunk 23]
```diff
@@ -281,12 +180,11 @@
...
         </profile>
         <profile>
-            <!-- Payara 5.202+ - Rest Client 1.4 -->
-            <id>payara5.202+</id>
+            <id>Payara Enterprise</id>
             <activation>
                 <property>
                     <name>payara.version</name>
-                    <value>/5\.20[2-4].*/</value>
+                    <value>/^[5-9]\.[0-9]{1,2}\..*/</value>
```
Comment (Pandrex247) [inline MicroProfile-Rest-Client/tck-runner/pom.xml:187]:
```suggestion
                    <value>/^[5-9]\.[0-9]{1,2}\.[0-9].*/</value>
```

**Pull request merged**